File: python/tests/sample/samples.py
import inspect

import pathlib
from json import load


def get_samples():
    file_name: str = \
        inspect \
            .getmodule(inspect.stack()[1][0]) \
            .__file__ \
            .split('/test_')[-1] \
            .replace('_', '-') \
            .split('.py')[0] + \
        '.json'

    root_dir: str = 'testing'
    if pathlib.Path().resolve().name.startswith('problems'):  # ran locally
        root_dir = '../../../testing'
    with open(
        f'{root_dir}/res/src/main/resources/{file_name}',
        'r',
        encoding='UTF-8',
    ) as json:
        samples = load(json)
        for sample in samples:
            sample['explanation'] = '\n'.join(sample['explanation'])
        return samples

File: testing/js/src/sample/samples.js
import fs from 'fs';
import {resolve, dirname} from 'path';
import {fileURLToPath} from 'url';

export default function getSamples(sampleFilename) {
  if (!sampleFilename) {
    let parts = _getCallerFile().split('/');
    let filename = parts[parts.length - 1];
    parts = filename.split('.');
    sampleFilename = `${parts[0]}.json`;
  }

  let samples =
    JSON.parse(
      fs.readFileSync(
        resolve(
          dirname(fileURLToPath(import.meta.url)),
          `../../../res/src/main/resources/${sampleFilename}`,
        ),
        'utf8',
      ),
    );
  for (let sample of samples) {
    sample.message = sample.explanation.join('\n');
  }
  return samples;
}

/**
 * @see {@link https://stackoverflow.com/questions/16697791/nodejs-get-filename-of-caller-function}
 */
function _getCallerFile() {
  let filename;
  let _pst = Error.prepareStackTrace;
  Error.prepareStackTrace =
    function(err, stack) {
      return stack;
    };
  try {
    let err = new Error();
    let callerfile;
    let currentfile;
    currentfile = err.stack.shift().getFileName();
    while (err.stack.length) {
      callerfile = err.stack.shift().getFileName();
      if (currentfile !== callerfile) {
        filename = callerfile;
        break;
      }
    }
  } catch (e) {
    console.error(e);
  }
  Error.prepareStackTrace = _pst;
  return filename;
}
